### ISSUE TITLE
crashes when npm updated package.json when installing dependencies

### ISSUE THREAD

**Issue by catdad** [catdad/electronmon#50]
**Steps**:
* use an app that watched `package.json` (for the main thread)
* `npm install -S <any package>`

See crash when npm updates package.json:

```bash
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at ChildProcess.target.send (internal/child_process.js:628:16)
    at EventEmitter.watcher.on (D:\Git\all-compare\node_modules\electronmon\src\electronmon.js:177:21)
    at EventEmitter.emit (events.js:182:13)
    at stat.then.stat (D:\Git\all-compare\node_modules\watchboy\index.js:196:28)
Emitted 'error' event at:
    at process.nextTick (internal/child_process.js:632:35)
    at process._tickCallback (internal/process/next_tick.js:61:11)
```

**Comment by catdad** [comment]
Turns out that this is because both `package.json` and `package-lock.json` are updated at the same time. `package.json` is detected as a main-thread full reset, the app exits, but before it is started up again, we try to send another message to the hook for a reload of the renderer thread. Since stdio is closed, it throws an error.

We should prevent sending further commands while the app is restarting.

**Comment by catdad** [comment]
Published in 1.1.1
